api/tests/Spatie/Snapshots/Driver/ECampYamlSnapshotDriver.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Tests\Spatie\Snapshots\Driver;
+
+use PHPUnit\Framework\Assert;
+use Spatie\Snapshots\Driver;
+use Symfony\Component\Yaml\Yaml;
+
+class ECampYamlSnapshotDriver implements Driver {
+    public function serialize($data): string {
+        if (is_string($data)) {
+            return $data;
+        }
+
+        return Yaml::dump($data, PHP_INT_MAX);
+    }
+
+    public function extension(): string {
+        return 'yml';
+    }
+
+    public function match($expected, $actual) {
+        if (is_array($actual)) {
+            $actual = Yaml::dump($actual, PHP_INT_MAX);
```

Comment:
Should you use `$this->serialize` instead of `Yaml::dump` here?

Reply:
if i pass the array, then i would need that.
currently i only use string as input.

Reply:
No, you don't only accept string as input. You explicitly check whether `$actual` is an array. You have explicit logic here which handles non-string inputs. Am I missing something completely?

Reply:
If i pass an array to serialize, Yaml::dump will also be used without the arguments.
This is just a copy of the existing driver without reserializing a string in the serialize method.
Depending on the decision in https://github.com/ecamp/ecamp3/pull/3943#discussion_r1358409631, i will use the same parameters in both cases (if they are more than the default)

Reply:
We'll leave it as it is, because this class is copied and simplified from https://github.com/spatie/phpunit-snapshot-assertions/blob/main/src/Drivers/YamlDriver.php and it's easier this way to keep it in sync with the upstream class.